src/core/Slider/Slider.test.tsx

```diff
@@ -0,0 +1,934 @@
+    container.querySelector('.iui-slider-component-container'),
+  ).toBeTruthy();
+  expect(container.querySelectorAll('.iui-slider-track').length).toBe(1);
+});
+// segments 0--10--20--30--40-----100
+it('should render odd tracks based in even number of values and `auto` trackDisplayMode', () => {
+  const { container } = render(<Slider values={[10, 20, 30, 40]} />);
+  assertBaseElement(container);
+  expect(
+    container.querySelector('.iui-slider-component-container'),
+  ).toBeTruthy();
+  // segments 10-20, 30-40
+  expect(container.querySelectorAll('.iui-slider-track').length).toBe(2);
+});
+it('should render 3 even track `even-segments` trackDisplayMode', () => {
+  const { container } = render(
+    <Slider trackDisplayMode='even-segments' values={[10, 20, 30, 40]} />,
+  );
+  assertBaseElement(container);
+  expect(
+    container.querySelector('.iui-slider-component-container'),
+  ).toBeTruthy();
+  // segments 0-10,20-30,40-100
+  expect(container.querySelectorAll('.iui-slider-track').length).toBe(3);
+});
+it('should render 2 odd track `even-segments` trackDisplayMode', () => {
+  const { container } = render(
+    <Slider trackDisplayMode='odd-segments' values={[10, 20, 30, 40]} />,
+  );
+  assertBaseElement(container);
+  expect(
+    container.querySelector('.iui-slider-component-container'),
+  ).toBeTruthy();
+  // segments 10-20, 30-40
+  expect(container.querySelectorAll('.iui-slider-track').length).toBe(2);
+});
+it('should not render track', () => {
+  const { container } = render(
+    <Slider trackDisplayMode='none' values={defaultSingleValue} />,
+  );
+  assertBaseElement(container);
+  expect(
+    container.querySelector('.iui-slider-component-container'),
+  ).toBeTruthy();
+  expect(container.querySelectorAll('.iui-slider-track').length).toBe(0);
+});
+it('should render custom tick marks', () => {
+  const { container } = render(
+    <Slider
+      values={defaultSingleValue}
+      tickLabels={<span className='custom-tick-mark'>Custom</span>}
+    />,
+  );
+  assertBaseElement(container);
+  expect(
+    container.querySelector('.iui-slider-component-container'),
+  ).toBeTruthy();
+  expect(container.querySelector('.custom-tick-mark')).toBeTruthy();
+});
+
+it('should activate thumb on pointerDown', () => {
+  let element: HTMLDivElement | null = null;
+  const onRef = (ref: HTMLDivElement) => {
+    element = ref;
+  };
+  const wrapper = render(<Slider ref={onRef} values={defaultSingleValue} />);
+  const { container } = wrapper;
+  assertBaseElement(container);
+  expect(element).toBeTruthy();
+  const thumb = container.querySelector('.iui-slider-thumb') as HTMLDivElement;
+  expect(thumb).toBeTruthy();
+  expect(thumb.classList.contains('iui-active')).toBeFalsy();
+
+  act(() => {
+    fireEvent.pointerDown(thumb, {
+      pointerId: 5,
+      buttons: 1,
+      clientX: 510,
+    });
+  });
+  expect(thumb.classList.contains('iui-active'));
+});
+
+it('focused thumb should process keystrokes', () => {
```

Comment:
all tests start with 'should', so would be great to change this naming too.. Something like `should process keystrokes, when thumb is focused`